### PR TITLE
Plane: tail-sitter only change shown AHRS View once transition is complete, and log transmission state

### PR DESCRIPTION
**Pull request by IamPete1**
Currently the ATT log switched on mode change, this mean that you get a inaccurate log due to Euler angles at 90 deg. 
ATT before

This chances to switch when the transition is complete, by default this is at 45 deg (or after a timeout). This avoids any euler angle funny business (and looks neater)
image
image

This also makes the same change for the attitude reported over MAVLink, this takes a little more getting used to But is, I think, a more intuitive way to understand what is happening in the transition.

This also adds logging of the transmission state to the QTUN message. This is usefull for AUTO mode where the it is much less clear what is going on.

image

### CODE REVIEW

File: ArduPlane/quadplane.cpp [hunk 1]
```diff
@@ -3380,3 +3381,23 @@ bool QuadPlane::in_vtol_land_sequence(void) const
 {
     return in_vtol_land_approach() || in_vtol_land_descent() || in_vtol_land_final();
 }
+
+// return true if we should VTOL view
```
Comment (kd0aij) [inline ArduPlane/quadplane.cpp:3385]: should **show** VTOL view

**Review by kd0aij — APPROVED** [review]
LGTM; definitely an improvement in display of euler angles

File: ArduPlane/quadplane.cpp (hunk 1 above)
Comment (IamPete1) [reply ArduPlane/quadplane.cpp:3385]: Thanks, fixed

**Comment by IamPete1** [comment]
I have added another commit to this that stops logging the copter target angles in Qasssit, instead using the plane ones.

The plane angle targets are what are driving the copter rate targets, so its better to use them directly than to try and re-construct out of rates.

Copter yaw rate is driven by plane roll as there is no desired yaw in plane, this can cause strange stuff, 
picturemessage_e53oplnw 2fp

Basically we should be logging the matching targets and actual, currently we do not.

File: ArduPlane/Log.cpp [hunk 2]
```diff
@@ -8,14 +8,7 @@ void Plane::Log_Write_Attitude(void)
...
     targets.y = nav_pitch_cd;
-
-    if (quadplane.in_vtol_mode() || quadplane.in_assisted_flight()) {
-        // when VTOL active log the copter target yaw
-        targets.z = wrap_360_cd(quadplane.attitude_control->get_att_target_euler_cd().z);
-    } else {
-        //Plane does not have the concept of navyaw. This is a placeholder.
-        targets.z = 0;
-    }
+    targets.z = 0; //Plane does not have the concept of navyaw. This is a placeholder.
```
Comment (kd0aij) [inline ArduPlane/Log.cpp:11]: The commit message for the existing code says "makes analysis of takeoffs easier"
I guess that's no longer true?

Comment (IamPete1) [reply ArduPlane/Log.cpp:11]: as far as I can tell the copter target is just nonsense in Qassist, i'm not sure how it would have helped with takeoffs.

In full Q modes we overwrite that value lower down anyway.

File: ArduPlane/quadplane.cpp [hunk 3]
```diff
@@ -3380,3 +3381,23 @@ bool QuadPlane::in_vtol_land_sequence(void) const
 {
     return in_vtol_land_approach() || in_vtol_land_descent() || in_vtol_land_final();
 }
+
+// return true if we should show VTOL view
+bool QuadPlane::show_vtol_view() const
+{
+    bool show_vtol = in_vtol_mode();
+
+    if (is_tailsitter()) {
```
Comment (tridge) [inline ArduPlane/quadplane.cpp:3390]: if disarmed then switch immediately

**Review by tridge — CHANGES_REQUESTED** [review]

File: ArduPlane/quadplane.cpp (hunk 3 above)
Comment (IamPete1) [reply ArduPlane/quadplane.cpp]: done and re-tested view switch

**Review by tridge — APPROVED** [review]
